### PR TITLE
Test: port some of delta-merge test into ut, part 1

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: ref #4609 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by JaySon-Huang — COMMENTED** [review]
It is a great start!

I think `tests/delta-merge-test/query/data_type` can also be rewritten by current UT framework.

**Comment by ywqzzy** [comment]
> It is a great start!
> 
> I think `tests/delta-merge-test/query/data_type` can also be rewritten by current UT framework.

Yes, almost every tests in /delta-merge-test/query will be ported to UT.
In the porting process, I will make the UT framework more easy to use.

**Review by Lloyd-Pottiger — APPROVED** [review]
Excellent work!

File: dbms/src/Debug/dbgFuncCoprocessor.cpp [hunk 1]
```diff
@@ -994,10 +994,14 @@ std::tuple<QueryTasks, MakeResOutputStream> compileQuery(
         };
 
     /// finalize
-    std::unordered_set<String> used_columns;
-    for (auto & schema : root_executor->output_schema)
-        used_columns.emplace(schema.first);
-    root_executor->columnPrune(used_columns);
+    if (!context.isExecutorTest())
```
Comment (SeaRise) [inline dbms/src/Debug/dbgFuncCoprocessor.cpp:997]: Why this change?

File: tests/delta-merge-test/query/expr/round_with_frac.test [hunk 2]
```diff
@@ -1,466 +0,0 @@
...
-=> DBGInvoke __raft_insert_row(default, i32, 3, 12, NULL)
-
-=> DBGInvoke __put_region(4, 0, 100, default, u32)
-=> DBGInvoke __raft_insert_row(default, u32, 4, 1, 0)
-=> DBGInvoke __raft_insert_row(default, u32, 4, 2, 4)
-=> DBGInvoke __raft_insert_row(default, u32, 4, 3, 5)
-=> DBGInvoke __raft_insert_row(default, u32, 4, 4, 499999999)
-=> DBGInvoke __raft_insert_row(default, u32, 4, 5, 500000000)
-=> DBGInvoke __raft_insert_row(default, u32, 4, 6, 4294967295)
-=> DBGInvoke __raft_insert_row(default, u32, 4, 7, null)
```
Comment (SeaRise) [inline tests/delta-merge-test/query/expr/round_with_frac.test:88]: I have a little doubt, after the migration, the relevant logic of the storage layer will not be able to go.
Is this expected?

Comment (ywqzzy) [reply tests/delta-merge-test/query/expr/round_with_frac.test:88]: > I have a little doubt, after the migration, the relevant logic of the storage layer will not be able to go. Is this expected?

Yes, IMO all the migrated tests don't need to use storage layer.

File: dbms/src/Debug/dbgFuncCoprocessor.cpp (hunk 1 above)
Comment (ywqzzy) [reply dbms/src/Debug/dbgFuncCoprocessor.cpp:997]: > Why this change?

Since in executor test, we don't call initOutputInfo() in DagContext.

Comment (SeaRise) [reply dbms/src/Debug/dbgFuncCoprocessor.cpp:997]: OK, can add some comments here

File: tests/delta-merge-test/query/expr/round_with_frac.test (hunk 2 above)
Comment (JaySon-Huang) [reply tests/delta-merge-test/query/expr/round_with_frac.test:88]: These tests under `/query/{executor,expr}` should focus on the compute layer and not care about the storage layer.

File: dbms/src/Flash/tests/gtest_query_executor.cpp [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+                "select * from default.test5 order by col_2 limit 2"));
+    }
+
+    {
+        auto cols = {toNullableVec<String>({{}, "test1"}), toNullableVec<Int64>({{}, 666})};
+        ASSERT_COLUMNS_EQ_UR(
+            cols,
+            executeRawQuery(
+                "select * from default.test5 order by col_2 limit 2"));
+    }
```
[the remark this answers is not in the document]
Comment (JaySon-Huang) [reply dbms/src/Flash/tests/gtest_query_executor.cpp:178]: Should it port this case? https://github.com/pingcap/tiflash/blob/a57e4b1fb3d15046f81289ab4d5e80c4d585f2c3/tests/delta-merge-test/query/executor/topn.test#L38-L43

File: tests/delta-merge-test/query/expr/logical_op.test [hunk 4]
```diff
@@ -12,27 +12,6 @@
...
-│           0 │           0 │           0 │          0 │          0 │          0 │           0 │           0 │           0 │       1 │       1 │
-│           0 │           0 │           0 │         \N │          0 │          0 │          \N │           0 │           0 │       1 │       1 │
-│          \N │           0 │           0 │          1 │          1 │         \N │          \N │           1 │          \N │      \N │       1 │
-│           0 │           0 │           0 │         \N │          0 │         \N │          \N │           0 │          \N │      \N │       1 │
-│          \N │           0 │           0 │         \N │          0 │         \N │          \N │           0 │          \N │      \N │       1 │
-└─────────────┴─────────────┴─────────────┴────────────┴────────────┴────────────┴─────────────┴─────────────┴─────────────┴─────────┴─────────┘
-
 => DBGInvoke __enable_schema_sync_service('true')
 
 => DBGInvoke __drop_tidb_table(default, test)
```
Comment (JaySon-Huang) [inline tests/delta-merge-test/query/expr/logical_op.test:17]: Why the test below is not ported?

Comment (ywqzzy) [reply tests/delta-merge-test/query/expr/logical_op.test:17]: > Why the test below is not ported?

The test below contains Decimal types which I am not pretty sure the executor UT framework can support it right now.
I will leave it to the next pr.

**Review by SeaRise — APPROVED** [review]
